src/ray/raylet/scheduling/cluster_resource_scheduler.cc

```diff
@@ -212,7 +218,7 @@ bool ClusterResourceScheduler::IsSchedulable(const ResourceRequest &resource_req
 
 int64_t ClusterResourceScheduler::GetBestSchedulableNode(
     const ResourceRequest &resource_request, bool actor_creation, bool force_spillback,
-    int64_t *total_violations, bool *is_infeasible) {
+    int64_t *total_violations, bool *is_infeasible, const TaskSpecification &task_spec) {
```

Comment:
Task spec shouldn't be exposed to cluster_resource_scheduler. Can you just pass blacklisted nodes directly here?